### DISCUSSION TITLE
Read and Recognize Zip Files

### DISCUSSION THREAD

**Discussion by Ralphman21** [advplyr/audiobookshelf#954] [Q&A]
I was previously using Calibre Web to access my audiobooks. To do so, I had to package books with multiple files into zip folders. It doesn't look like the scan of Audiobookshelf picks up and recognizes these. Is this something that could be done in the future?

**Comment by advplyr** [comment]
I don't think this will be supported in the future, but I don't want to say it is impossible. We would have to extract the zip files in the folder they are in and it just becomes a mess so it's very unlikely we do this.

**Reply by bilogic** [reply]
1. Why not treat the zip the same way when a folder is uploaded and document the expected ZIP structure?
2. And link to the documentation right from the upload page
3. I often generate a ZIP from a source that is directly compatible with ABS for uploading, i.e. audio book shows up perfectly
4. The only down side is, I always have to unzip the it first, then upload the `bilogic` folder in ABS, see below

### Zip file structure
```
└── bilogic
    └── book-title
        ├── metadata
        |   ├── cover.jpg
        |   └── metadata.json
        ├── part-001.mp3
        └── part-002.mp3
```

**Comment by JamesM92** [comment]
I would also like this feature, it does mean that the server would have to handle unzipping and handling the content, but the server already handles whatever we through in a folder for it to parse, this would just add an additional processing step.

I think it would be very beneficial to not have to unzip a large .zip file to upload it. It would take longer to process of course but i dont think thats a terrible thing.

**Comment by susan3333333** [comment]
VLC does this, I used to use it, I think you just played them as-is as far as I know it didn't outright unzip them. But this was a long time ago so I am not really sure how it worked.
